gesasso/frontend/src/components/RequestForm/RequestContext/modules/OnsiteSaleContext.tsx

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) Cesar Richard 2023.
+ */
+
+import React, { useState } from 'react';
+import { Input } from 'semantic-ui-react';
+// eslint-disable-next-line import/no-cycle
+import { RequestContextComponentProps } from '@gesasso/components/RequestForm/RequestContext/modules/RequestContext';
+
+const OnsiteSaleContext = ({ context, onContextChange }: RequestContextComponentProps): React.ReactElement<any> => {
+  const [contextState, setContextState] = useState(context);
+
+  const handleChange = (e) => {
+    setContextState(e);
+    onContextChange(e);
+  };
+
+  return (
+    <div>
+      <Input placeholder="Context" onChange={(e) => handleChange(e.target.value)} value={contextState} />
+    </div>
+  );
+};
```

Comment:
Pour éviter un state inutile et un re-render à chaque caractère
```suggestion
const OnsiteSaleContext = ({ context, onContextChange }: RequestContextComponentProps): JSX.Element => {
  const inputRef = useRef(null);

  const handleChange = (e) => {
    onContextChange(inputRef.current.value);
  };

  return (
    <div>
      <Ref innerRef={inputRef}>
        <Input placeholder="Context" onChange={handleChange} />
      </Ref>
    </div>
  );
};
```

Reply:
+ peut-être en faire un custom hook, plutôt qu'avoir une palanquée de fichier dupliqué par context ?

Reply:
Apres ces fichiers la sont juste des placeholder , pour "avoir qqch" pour tester en attendant les vrais forms